shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala

```diff
@@ -404,6 +446,7 @@ class UCX(executorId: Int, usingWakeupFeature: Boolean = true) extends AutoClose
         val is = socket.getInputStream
 
         // "this executor id will receive on tmpLocalReceiveTag for this Connection"
+        ucxWorkerAddress.rewind()
```

Comment:
How do you exchange management port: https://github.com/NVIDIA/spark-rapids/blob/branch-0.6/shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala#L203 ?

Reply:
lazy val is [thread safe](https://stackoverflow.com/a/23048497/3271168). Why need def?

Reply:
We do this as part of the optional `topologyConf` that is in every `BlockManagerId`.

https://github.com/NVIDIA/spark-rapids/blob/branch-0.6/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala#L180

Reply:
> lazy val is thread safe. Why need def?

Yeah setting the lazy val is, so the reference is going to be set correctly. But mutating the `ByteBuffer` with `rewind` is not.

Reply:
Ok, this could be solved with `byteBuffer.duplicate()`: https://github.com/NVIDIA/spark-rapids/pull/2131/commits/6d2bdb5f77cd3f4e7a053b44c9a4befe09850d1a

Reply:
Yeah, could also use `asReadOnlyBuffer()`. So perhaps a lazy val that does the heavy lifting, and a def that only returns "read only" or "duplicate"?

Reply:
Since "the two buffers' position, limit, and mark values will be independent.", we should be able to remove the `rewind()` call, or do that once at `lazy val` time.